### PR TITLE
Add functionality to handle missing values in HubDB columns

### LINKED ISSUE #84: Make app more robust (so that missing values from HubDB do not cause unexpected behavior)
Currently, when values are missing from certain columns in the Events HubDB table, the app displays some unexpected behaviors. The app also lacks any warnings, if there are important values missing (eg. the event's title). 

I've listed the columns with issues below and some suggestions on how to address them. 

NO WARNINGS 

- [x] Page title: There are no warnings on the page or logged to the console, if this value is missing. I'd suggest that we add a warning to the console.  

- [x] Page path: There are no warnings on the page or logged to the console, if this value is missing. 
I'd suggest we add a warning to the console here in particular, because the user cannot select the event card to navigate to the event details page without this value. 

- [x] Name: There are no warnings or default value on the page, if the event doesn't have a title. I'd suggest we had some logic to the EventCard component to set a default value here (ie. "Untitled Event").

- [x] Event Description: There are no warnings or default value on the page, if the event doesn't have a title. I'd suggest we had some logic to the EventDetails component to set a default value here (ie. "Event Description Missing").

UNEXPECTED BEHAVIOR: 

- [x] Start: If an event is missing a value for "Start", that card simply does not appear on the landing page. But if no columns have a start time, then all three events will appear on the page, and their start times are the current date and time. I'd suggest that if the start time is missing, we add a warning to the page, ie. "The start time has not been determined." 

- [x] End: If the event is missing a value for "End", the end time for the event is set to the current date and time, I'd suggest that we simply omit the end time in the EventDetails component. 

- [x] Event Capacity: If there is a missing value, the EventSpacesLeft component displays "There are NaN spots left." I'd suggest that we add some logic to this component to address this. 

- [x] Registered Attendee Count: If there is a missing value, the EventSpacesLeft component displays "There are NaN spots left." I'd suggest that we add some logic to this component to address this.

### PR DESCRIPTION
**Pull request by kemmerle**
Fixes #84 

I have four main concerns at the moment: 

1) I've added a function in `AppContext.js` to filter events and to add default values, where values are missing. I've taken advantage of the lodash `defaults` method to do so, but I'm curious whether anyone else might have other ideas how to achieve this.

2) I have added default values for certain columns--do these defaults make sense to everyone?  

DEFAULTS SET: 
Name: 'Untitled event' 
Event Description: 'Missing event description'
Event Capacity: 25
Registered Eventee Count: 0
Attendance Type: Virtual 

3) Currently, if an event is missing the Page Title or Page Path columns, no warnings appear. Without the Page Path column in particular, the user cannot select the event card to navigate to the event details page without this value. I'm not quite sure the best way to deliver a warning to the user. I'd appreciate any feedback. 

4) I'd like to set default values for the Start and End columns, but I'm unsure what the best values would be. It seems unwise to choose an arbitrary time and date. Again, I'd appreciate any feedback. 

cc @gcorne @anthmatic @levinkeo

### CODE REVIEW

**Comment by anthmatic** [comment]
3. If a row is missing key info like `path`, maybe we should just not show that event.

4. I'm not sure that fallback values would really work here. It would he helpful. I agree that arbitrary times wouldn't make sense. What if missing start or end times mean that the event is an "all day" event?

> I'm not quite sure the best way to deliver a warning to the user.

My two thoughts on this are:
- Display warnings or errors only in the page editor where the user might see them.
- Print warnings to console, although this is more helpful to developers than marketers and would be visible to all unless we add a `debug` flag.

**Comment by kemmerle** [comment]
I'd like an extra set of eyes on this PR before I merge, since the changes are across the app. I'm listing below how I've handled missing values for specific HubDB table columns:

1) I've set default values for the following columns:  
Name
Event Description
Event Capacity
Registered Attendee Count

2) I'm filtering missing values for the start/end columns. 

I decided that it didn't make a lot of sense to set arbitrary default values for the start/end times. There are use cases where someone won't know the date/time before they create the event. Now if a start time value is missing from a HubDB table, then "Time to be determined" is displayed in the `EventCard` component. If the start or end time value are omitted, then they simply isn't displayed in the `EventDetails` component.

3) I created a component to display a warning in the page editor if the page title or page path are missing. 

Currently, the `ErrorBoundary` component handles errors in the app, but this is a Class component (and it needs to be to take advantage of functions like `getDerivedStateFromError` and `componentDidCatch`. See: https://reactjs.org/docs/error-boundaries.html). 

To display errors about missing HubDB columns in a component, I needed to pass down the events array as a prop, so that we can check for missing values. I created a separate component for this, but do let me know if I can accomplish this elsewhere and streamline the app a bit.

File: src/components/ErrorPageEditor.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+const ErrorPageEditor = ({ events, portalId, tableId }) => {
+  function renderErrors() {
+    if (events) {
+      if (
```
Comment (anthmatic) [inline src/components/ErrorPageEditor.js:7]: can these two if statements get merged into one?

**Comment by anthmatic** [comment]
> To display errors about missing HubDB columns in a component, I needed to pass down the events array as a prop, so that we can check for missing values.

I think in general we could add error boundaries around components (like the card component) which might be worth a look. BUT, I like the example of adding an in-editor warning and I think that's useful to keep around. Might even be worth explaining that somewhere (comment or readme)

**Pull request merged**